pyproject.toml

```
@@ -1,7 +1,29 @@
+[tool.mypy]
+pretty = true
+ignore_missing_imports = false
+check_untyped_defs = true
+no_implicit_reexport = false
+disallow_untyped_defs = false # remove once the project is typed
+strict = true
+plugins = ["pydantic.mypy"]
+
+[[tool.mypy.overrides]]
+module = ["pandas", "sklearn.*", "joblib", "scipy.*", "mariadb"]
+ignore_missing_imports = true
+
+[tool.pyright]
+# used in Pylance extension for language server
+# type check is done by mypy, disable to avoid unwanted errors
+typeCheckingMode = "off"
+
 [tool.isort]
 profile = "black"
 
+[tool.ruff]
+line-length = 100
+
 [tool.ruff.lint]
 ignore = [
     "F841",  # don't complain about unused variables
 ]
+select = ["RUF", "FURB", "PL"] # suggestion: add PTH
```

These are quite strict, and some are questionable too (e.g. FURB140).

At this point I would focus attention on the bigger issues (code structure) instead of small details such as variable naming.
```suggestion
select = ["RUF"]
```

---

removed